tests/subsys/pm/power_domain/src/main.c

```diff
@@ -279,5 +279,15 @@ ZTEST(power_domain_1cpu, test_power_domain_device_balanced)
 	zassert_equal(state, PM_DEVICE_STATE_ACTIVE);
 }
 
+ZTEST(power_domain_1cpu, test_on_power_domain)
+{
+	zassert_true(device_is_ready(domain), "Device is not ready!");
+	zassert_true(device_is_ready(deva), "Device is not ready!");
+	pm_device_power_domain_remove(deva, domain);
+	zassert_false(pm_device_on_power_domain(deva), "devc is in the power domain.");
+	pm_device_power_domain_add(deva, domain);
+	zassert_true(pm_device_on_power_domain(deva), "deva is not in the power domain.");
+}
```

Comment:
would you mind to similar flow for "devc" as well ? `devc` is not declared in the DT, so would be nice to check that it behaves the same way too.

Reply:
I added similar flow for device C.